src/OrchardCore.Modules/OrchardCore.Alias/Indexes/AliasPartIndex.cs

```diff
@@ -32,11 +32,11 @@ public AliasPartIndexProvider(IServiceProvider serviceProvider)
         public override void Describe(DescribeContext<ContentItem> context)
         {
             context.For<AliasPartIndex>()
+                .When(contentItem => (contentItem.IsPublished() || contentItem.Latest) && contentItem.Has<AliasPart>())
```

Comment:
So the problem with using `.When` here is twofold.
`.When` stops the deletes happening from the Index (that's its purpose).

so if you use it to do `contentItem.IsPublished()` then when a content item is not published (but was previously), it's previous record will not be deleted from the Index, so will remain, and will still have the `Published` value set to true.

The Alias should also be covered by the `Remove()` concept we introduced (seperate issue. didn't think about that one as well)

Reply:
Yes, it's only a first pass, i will try thes use cases tomorrow

> The Alias should also be covered by the Remove() concept we introduced (seperate issue. didn't think about that one as well)

Yes i saw it, see my comment in #8472